config/yamlcfg/yaml2proto_test.go

```diff
@@ -710,3 +712,103 @@ func Test_ReadConfig(t *testing.T) {
 		})
 	}
 }
+
+func Test_getDefaults(t *testing.T) {
+	tests := []struct {
+		name  string
+		paths []string
+		want  []string
+	}{
+		{
+			name: "empty paths",
+		},
+		{
+			name:  "simple case",
+			paths: []string{"foo/config.yaml", "foo/default.yaml"},
+			want:  []string{"foo/default.yaml"},
+		},
+		{
+			name:  "no defaults",
+			paths: []string{"foo/config.yaml"},
+		},
+		{
+			name:  "multiple defaults",
+			paths: []string{"foo/default.yaml", "bar/default.yaml"},
+			want:  []string{"foo/default.yaml", "bar/default.yaml"},
+		},
+		{
+			name:  "subdirs",
+			paths: []string{"foo/bar/default.yaml"},
+			want:  []string{"foo/bar/default.yaml"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := getDefaults(test.paths)
+			if diff := cmp.Diff(test.want, got); diff != "" {
+				t.Fatalf("returned with incorrect value, %v", diff)
+			}
+		})
+	}
+}
+
+func Test_PathDefault(t *testing.T) {
+	overallDefaults := DefaultConfiguration{
+		DefaultTestGroup: &config.TestGroup{
+			NumColumnsRecent: 5,
+		},
+	}
+	localDefaults := DefaultConfiguration{
+		DefaultTestGroup: &config.TestGroup{
+			NumColumnsRecent: 10,
+		},
+	}
+
+	tests := []struct {
+		name         string
+		path         string
+		defaultFiles map[string]DefaultConfiguration
+		defaults     DefaultConfiguration
+		want         DefaultConfiguration
+	}{
+		{
+			name: "empty works",
+		},
+		{
+			name: "basically works",
+			path: "foo/config.yaml",
+			defaultFiles: map[string]DefaultConfiguration{
+				"foo": localDefaults,
+			},
+			defaults: overallDefaults,
+			want:     localDefaults,
+		},
+		{
+			name: "path not in map uses overall defaults",
+			path: "config.yaml",
+			defaultFiles: map[string]DefaultConfiguration{
+				"foo": localDefaults,
+			},
+			defaults: overallDefaults,
+			want:     overallDefaults,
+		},
+		{
+			name: "path in subdirectory of path in map uses overall defaults",
```

Comment:
nit: "path in subdirectory of local default uses overall defaults"

It's important to me that this description be clear, since I might expect the default closest to the config file, not the root default.

Reply:
Done